src/tools.cpp

```diff
@@ -153,6 +153,50 @@ static std::string removeLastPathElement(const std::string& path,
   return newPath;
 }
 
+namespace
+{
+
+std::vector<std::string> split(const std::string & str, char delim)
+{
+  std::vector<std::string> result;
+  auto start = str.begin();
+  while ( true ) {
+      const auto end = find(start, str.end(), delim);
+
+      result.push_back(std::string(start, end));
+      if ( end == str.end() )
+        break;
+      start = end + 1;
+  }
+
+  return result;
+}
+
+} // unnamed namespace
+
+std::string
+computeRelativePath(const std::string& path1, const std::string& path2)
+{
+    const auto a = split(path1, '/');
+    const auto b = split(path2, '/');
+    const auto l = std::min(a.size(), b.size());
+
+    const auto x = mismatch(a.begin(), a.begin() + l, b.begin());
+
+    const size_t ups = (a.end() - x.first) - 1;
+    std::string r;
+    for ( size_t i = 0; i < ups; ++i ) {
+      r += "../";
+    }
+
+    for ( auto it = x.second; it != b.end(); ++it ) {
+        if ( it != x.second )
+          r += "/";
+        r += *it;
+    }
+    return r;
+}
```

Comment:
kiwix-lib already have a `computeRelativePath` (https://github.com/kiwix/kiwix-lib/blob/master/src/tools/pathTools.cpp#L176-L199).
The kiwix one does a bit more as it normalize the path (remove unnecessary internal `..`).
But this implementation is a bit simpler (use of `mismatch` instead of reimplement it)
It would be nice to have some synchronization (or even move the `computeRelativePath` in libzim and use only this one)

I'm not sure this implementation can handle windows path (but we are looking for `/` in the path elsewhere in the code, to the questioning is more global to all zimdump tool)

Reply:
I don't think anything needs to be done about it in this PR.